### PR TITLE
Bug 2106476: Preserve iface ordering in ptpconfig

### PR DESCRIPTION
**Pull request by josephdrichard**

### CODE REVIEW

File: pkg/daemon/config.go [hunk 1]
```diff
@@ -107,35 +108,42 @@ func tryToLoadOldConfig(nodeProfilesJson []byte) ([]ptpv1.PtpProfile, bool) {
...
+				globalIsDefined = true
+			}
+			currentSection = ptp4lConfSection{options: map[string]string{}, sectionName: currentSectionName}
+		} else if currentSectionName != "" {
 			split := strings.IndexByte(line, ' ')
 			if split > 0 {
-				section := output.sections[currentSection]
-				section.options[line[:split]] = line[split:]
-				output.sections[currentSection] = section
+				currentSection.options[line[:split]] = line[split:]
```
Comment (aneeshkp) [inline pkg/daemon/config.go:136]: So the section are added to an array, that way section is guaranteed to be in order but the content are not, correct ?

Comment (josephdrichard) [reply pkg/daemon/config.go:136]: correct.  There is no requirement for the order within a section to be kept.

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**